File: benchmarks/python/list_functional.py
import sys
from functools import reduce

n = int(sys.argv[1]) if len(sys.argv) > 1 else 10_000

items = list(range(n))
evens = list(filter(lambda x: x % 2 == 0, items))
squared = list(map(lambda x: x * x, evens))
total = reduce(lambda a, b: a + b, squared, 0)

print(total)

File: benchmarks/node/list_functional.js
"use strict";

const n = process.argv[2] !== undefined ? parseInt(process.argv[2], 10) : 10_000;

const items = [];
for (let i = 0; i < n; i++) items.push(i);

const evens = items.filter((x) => x % 2 === 0);
const squared = evens.map((x) => x * x);
const total = squared.reduce((a, b) => a + b, 0);

console.log(total);
